File: src/main/java/net/frozenblock/configurableeverything/entity/util/zombie/ai/ZombieFleeSunGoal.java
package net.frozenblock.configurableeverything.entity.util.zombie.ai;

import net.frozenblock.configurableeverything.config.EntityConfig;
import net.frozenblock.configurableeverything.config.MainConfig;
import net.minecraft.world.entity.PathfinderMob;
import net.minecraft.world.entity.ai.goal.FleeSunGoal;

public class ZombieFleeSunGoal extends FleeSunGoal {

	public ZombieFleeSunGoal(PathfinderMob mob, double speedModifier) {
		super(mob, speedModifier);
	}

	@Override
	public boolean canUse() {
		if (!(MainConfig.get(false).entity && EntityConfig.get(false).zombie.zombiesAvoidSun)) {
			return false;
		}
		return super.canUse();
	}
}

File: src/main/java/net/frozenblock/configurableeverything/entity/util/zombie/ai/ZombieFleeSunGoal.kt
package net.frozenblock.configurableeverything.entity.util.zombie.ai;

import net.frozenblock.configurableeverything.config.EntityConfig;
import net.frozenblock.configurableeverything.config.MainConfig;
import net.minecraft.world.entity.PathfinderMob;
import net.minecraft.world.entity.ai.goal.FleeSunGoal;

class ZombieFleeSunGoal(mob: PathfinderMob, speedModifier: Double) : FleeSunGoal(mob, speedModifier) {

	override fun canUse(): Boolean {
		return if (!(MainConfig.get().entity && EntityConfig.get().zombie.zombiesAvoidSun))
			false
		else super.canUse()
	}
}

